### DISCUSSION TITLE
How to get user ip?

### DISCUSSION THREAD

**Discussion by lizheming** [deta/space-docs#38] [Q&A] [answered]
I output all request headers into https://waline-1-j2196449.deta.app/ . It seems `x-real-ip` is what I find, but it's the deta host ip, not visitor user's ip.

**Comment by abdelhai** [comment]
AWS Global Accelerator, which we use, unfortunately, doesn't preserve the client's IP address.
there's a way to fix this by using a custom domain. lmk if this would be a workable solution, so I could share the details.

**Reply by lizheming** [reply]
https://waline-dev.deta.lithub.cc/ I'm afraid it doesn't work. `x-forward-for` and `x-real-ip` is the same aws ip

**Reply by abdelhai** [reply]
it requires the use of a cloudflare proxy. we will provide a guide soon. is this a must-have for Waline?

**Reply by lizheming** [reply]
It doesn't have to be. It’s just that this one is a problem. There was a [discussion](https://github.com/orgs/deta/discussions/282) that reported this problem before, but it was closed recently, so I tested it and feedback again.

**Comment by lihugang** [comment, marked as answer]
Here's a guide of accessing client ip address.

https://deta.space/docs/en/guides/accessing-client-ip-address

**Reply by abdelhai** [reply]
oh yeah, forgot to share. thanks for posting!
we wrote it specifically as a response to @lizheming

**Reply by lizheming** [reply]
@abdelhai Thanks a lot for the reply, I want to make sure so I have to use Cloudflare as DNS to get the user IP right?

**Reply by abdelhai** [reply]
pretty sure (as a DNS & Proxy). Not sure if others have similar features, Fastly should also have the same feature.

i'm aware this is very inconvenient, and we would like to just support IP preservation natively, it's just a huge undertaking. we're actively hiring, so we could implement these things quicker.

**Reply by lizheming** [reply]
OK, thank you very much, looking forward to realizing it soon~
